pyigm/field/igmfield.py

```diff
@@ -0,0 +1,236 @@
+        # coord
+        if isinstance(radec, (tuple)):
+            self.coord = SkyCoord(ra=radec[0], dec=radec[1])
+        elif isinstance(radec, SkyCoord):
+            self.coord = radec
+
+        # Field
+        self.name = name
+
+        # Cosmology
+        if cosmo is None:
+            from astropy.cosmology import WMAP9 as cosmo
+            if verbose is True:
+                print('IgmGalxyField: Using WMAP9 cosmology')
+        self.cosmo = cosmo
+
+        # Init
+        self.igm = None
+        self.targets = None
+        self.galaxies = None
+        self.observing = None
+        self.selection = None
+
+    def calc_rhoimpact(self, obj, los_coord=None):
+        """Calculate impact parameter from LOS RA/DEC for a set of objects
+
+        Parameters
+        ----------
+        obj : Table
+          (can be anything that takes 'RA','DEC' and 'Z')
+          Sources for calculation
+        los_radec : SkyCoord, optional
+          Defaults to field RA/DEC
+
+        Returns
+        -------
+
+        rho : Quantity (array usually)
+          Impact parameter(s) in kpc
+        """
+        if los_coord is None:
+            los_coord = self.coord
+        # Coord
+        o_coord = SkyCoord(ra=obj['RA']*u.deg, dec=obj['DEC']*u.deg)
+        ang_sep = o_coord.separation(los_coord).to('arcmin')
+        # Cosmology
+        kpc_amin = self.cosmo.kpc_comoving_per_arcmin(obj['Z'] ) # kpc per arcmin
+        rho = ang_sep * kpc_amin / (1+obj['Z']) # Physical
+        # Return
+        return rho
+
+    def get_associated_galaxies(self, z, los_coord=None, R=300*u.kpc, dv_tol=500*u.km/u.s):
+        """Return a Table of associated galaxies for a given redshift and separation
+
+        Parameters
+        ----------
+        z : float
+          Redshift for association (usually IGM/CGM absorber redshift)
+        R : Quantity
+          Radius of impact parameter for association [300kpc]
+        dv_tol : Quantity
+          Velocity window for association [500km/s]
+        los_coord : SkyCoord, optional
+          Line-of-sight coordinates
+
+        Returns
+        -------
+        assoc_gal : Table
+          Table of associated galaxies (if any)
+        rho : Quantity
+          Impact parameters [kpc]
+        """
+        # los_radec
+        if los_coord is None:
+            los_coord = self.coord
+        # Cut on z first
+        dv_gal = const.c.to('km/s') * (self.galaxies['Z']-z)/(1+z)  # Approximate
+        gdz = np.where(np.abs(dv_gal)<dv_tol)[0]
+        if len(gdz) == 0:
+            return None
+        #
+        gdz_gal = self.galaxies[gdz]
+        rho = self.calc_rhoimpact(gdz_gal, los_coord)  # Could add this to Table
+        #
+        gd_rho = np.where(rho < R)[0]
+        if len(gd_rho) == 0:
+            return None
+        # Return
+        return gdz_gal[gd_rho], rho[gd_rho]
+
+    def get_observed(self, theta, subtab=None):
```

Comment:
Explain subtab in doc string. Make theta optional, and use None as default (meaning retrieve all).